src/bootstrap/test.rs

```diff
@@ -1035,7 +1035,7 @@ impl Step for Compiletest {
         flags.push("-Zunstable-options".to_string());
         flags.push(builder.config.cmd.rustc_args().join(" "));
 
-        if let Some(linker) = builder.linker(target) {
+        if let Some(linker) = builder.linker(target, false) {
```

Comment:
Why is compiletest not allowed to use the linker? Can we get a comment here?

Reply:
We want to test that `rustc` works without explicitly passing it LLD.

Reply:
I don't really think that makes sense, if you've asked for LLD we should make everything use it, seems weird for tests to always opt-out (unless you actually pass `linker = lld`).

CI of course may want to opt out here eventually, but even that I'm not sure about.

I would prefer to not make this change for now though. Do we have concrete examples were LLD would have masked bugs or something like that?

Reply:
It makes sense to test what we ship, so unless `rustc` uses LLD by default, our tests should not either.

Reply:
It makes sense for CI to do so, buy I feel like requiring that locally is a bit odd? I am okay landing this with tests not using LLD as long as the docs for the flag mention it as well.